destination/dummy_dest.py

```diff
@@ -41,9 +44,13 @@ def main():
 def reply(conn_socket):
     try:
         while True:
-            msg = conn_socket.recv(2048)
-            mod_msg = msg.upper()
-            conn_socket.send(mod_msg)
+            msg = conn_socket.recv(2048).decode('utf-8')
+            print('Recieved: {}'.format(msg))
+            if msg == 'SYN': 
```

Comment:
Why is this handshake necessary?

Reply:
agree

Reply:
@DestructiveReasoning @stumash 

To clarify on the SYN/ACK:

I had to adhere to the method of `onion_router.py` of establishing the VC.
Because each router returns ACK on the first message and then waits for the second one to establish the next hop connection, I had to choose a dummy string to send as a first message.
Choosing SYN was the natural answer

`onion_router.py` line 63: 
```python
#(TODO:decrypt data, initialize+respond with symkey. for now send 'ACK')
back_conn.sendall("ACK".encode('utf-8'))
```

As to why it was also added to the destination node, it was to keep consistency throughout the configuration of the VC as you can see here:

`onion_client.py` line 102
```python
 def setup_vc(path, conn):
 # get ACK from first node
 conn.sendall(b'SYN')
 response = conn.recv(1024).decode('utf-8').rstrip()
 if response != 'ACK': return false
 if args.verbose: print('ACK recieved')

 # get ACK for remaining nodes and destination
 for i in range(1,4):
     setup_onion = encapsulate(path[i]['addr'], path[i]['key'], 'SYN')
     if args.verbose: print('setup_onion: {}'.format(setup_onion))
     
     # TODO encrypt data
     
     conn.sendall(json.dumps(setup_onion).encode('utf-8'))
     response = conn.recv(1024).decode('utf-8').rstrip()
      if response != 'ACK': return false
     if args.verbose: print('ACK recieved')
```

I am aware it won't allow accessing websites because they won't return `ACK`s, I chose to add because currently, we are only talking to internal nodes. But if issue #15 is expected to be solved before the project submission and thus allowing us to communicate to other ports (e.g. accessing websites) I can change it right now.

I'll wait on your decision

Reply:
I don't think this is necessary, because it's just not a realistic enough scenario in my opinion. The originator knows how many intermediate nodes there are, so can't he just wait for that many ACK's?

Reply:
that works, so to conclude:
- Recieve 3 `ACK`s because 3 intermediate nodes
- For the actual destination first package received is actual first data packet

When you confirm I'll apply the change

Reply:
yes cool

Reply:
Yeah that's what I meant. Sounds good

Reply:
Fixed in 31e617e7b16ab2a1f0db2407a3bcd6f22e4cab33